src/model_clay.py

```diff
@@ -281,6 +279,9 @@ def forward(self, datacube):
             cube
         )  # [B G L D] - patchify & create embeddings per patch
 
+        # Move position & band encoding to the device
+        self.pos_encoding = self.pos_encoding.to(patches.device)
+        self.band_encoding = self.band_encoding.to(patches.device)
```

Comment:
@srmsoumya, I feel like we can also remove the `.to(device)` call here?
```suggestion
```

Reply:
Ah, I think I found the core issue. The `pos_encoding` and `band_encoding` tensors are declared in the `__init__` method, before Lightning has managed to cast them to the correct device.

https://github.com/Clay-foundation/model/blob/ee74c91b99cbf9a7c304c0d37335c128d1ae7566/src/model_clay.py#L89-L95

Ideally, we'll need to define the device at this point. Let me see if there's a way to do that.

Reply:
Ok, went with using `self.register_buffers` at commit 25f83e62863da5e5ba4a0a8625a8ef8d1b2afd2a. Have set `persistent=False` since those pos_encoding and band_encoding tensors are fixed, and won't need to be saved to the `state_dict`.